### PR TITLE
Removing ContainerManifest

### PR DESCRIPTION
**Pull request by krousey**
#8087

cc @caesarxuchao @nikhiljindal @bgrant0607 

@dchen1107 Please double check pkg/kubelet/config/file_test.go and other kubelet changes.

### CODE REVIEW

File: pkg/api/validation/validation.go [hunk 1]
```diff
@@ -863,27 +863,6 @@ func validateContainers(containers []api.Container, volumes util.StringSet) errs
 
 var supportedManifestVersions = util.NewStringSet("v1beta1", "v1beta2")
```
Comment (caesarxuchao) [inline pkg/api/validation/validation.go]: This line can be removed as well.

File: pkg/api/serialization_test.go [hunk 2]
```diff
@@ -125,7 +125,7 @@ func TestList(t *testing.T) {
 	roundTripSame(t, item)
 }
 
-var nonRoundTrippableTypes = util.NewStringSet("ContainerManifest", "ContainerManifestList")
+var nonRoundTrippableTypes = util.NewStringSet()
 var nonInternalRoundTrippableTypes = util.NewStringSet("List", "ListOptions", "PodExecOptions")
```
Comment (caesarxuchao) [inline pkg/api/serialization_test.go]: @nikhiljindal @bgrant0607 , can we remove `nonRoundTrippableTypes`? Or are we expecting there will be new member in this sets in the future?

**Comment by dchen1107** [comment]
I only reviewed kubelet related pieces, that lgtm.

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (krousey) [inline pkg/api/validation/validation.go]: done

File: pkg/api/serialization_test.go (hunk 2 above)
Comment (caesarxuchao) [inline pkg/api/serialization_test.go]: cc @smarterclayton

File: pkg/api/validation/validation.go (hunk 1 above)
Comment (caesarxuchao) [inline pkg/api/validation/validation.go]: Thanks!

**Comment by krousey** [comment]
I'm OOO right now, but occasionally checking in on this one. Is there anything left that I need to do?

**Comment by caesarxuchao** [comment]
I'm still not sure about `nonRoundTrippableTypes`, but it shouldn't prevent this PR to be merged. Thank you. LGTM.

**Comment by bgrant0607** [comment]
I think it's fine to leave nonRoundTrippableTypes in the code for now.

**Comment by satnam6502** [comment]
Milestone?

**Comment by caesarxuchao** [comment]
@bgrant0607 could you set triage it? I don't think I have the right to set the milestone.

**Comment by bgrant0607** [comment]
This is ok to go in, but the test failures need to be fixed.

**Comment by krousey** [comment]
The only test failure I saw was what looked to be a flaky shippable build,
and I couldn't figure out how to rerun it. Did the e2e fail? I'm on my
phone and can't really see the test statuses.

**Comment by satnam6502** [comment]
@k8s-bot ok to test

**Comment by caesarxuchao** [comment]
Close and re-open the PR to trigger shippable

**Comment by krousey** [comment]
Rebased

File: contrib/mesos/pkg/executor/executor_test.go [hunk 3]
```diff
@@ -437,9 +437,9 @@ func TestExecutorStaticPods(t *testing.T) {
 		assert.NoError(t, err)
 		spod := `{
 	"apiVersion": "v1beta3",
-	"name": "%v",
 	"kind": "Pod",
 	"metadata": {
+		"name": "%v",
```
Comment (krousey) [inline contrib/mesos/pkg/executor/executor_test.go]: @jdef FYI I had to fix this error in Pod specification. By removing the name defaulting in pkg/kubelet/config/common.go, this test started failing.

**Comment by krousey** [comment]
I fixed the mesos test failure that relied on Pod name defaulting logic in kubelet.

**Comment by caesarxuchao** [comment]
Thanks @krousey. LGTM. @bgrant0607, could you give the second LGTM? Thanks.

File: contrib/mesos/pkg/executor/executor_test.go [hunk 4]
```diff
@@ -437,9 +437,9 @@ func TestExecutorStaticPods(t *testing.T) {
 		assert.NoError(t, err)
 		spod := `{
 	"apiVersion": "v1beta3",
```
Comment (bgrant0607) [inline contrib/mesos/pkg/executor/executor_test.go]: Could we just change this to v1 while we're touching it?

**Comment by bgrant0607** [comment]
LGTM. We can do the v1beta3->v1 conversion later, if necessary.

**Pull request merged**